Cargo.toml

```diff
@@ -123,6 +123,10 @@ lto = true
 panic = 'abort'
 codegen-units = 1
 
+[profile.test]
+opt-level = 3
```

Comment:
We should experiment with Cargo profile overrides to just optimize the math-heavy dependencies: https://docs.rust-embedded.org/book/unsorted/speed-vs-size.html#optimizing-dependencies

Reply:
It's not necessary to block making this change (which is also in #1603) on doing that. We can do so *as well* to further improve build+test times, but this is *already* a strict improvement on both the time and debugging QoL axes over `--release`, and you can override it with `--profile=dev` when you need to.

Reply:
Filed as #1606.